diff --git a/arch/arm64/boot/dts/marvell/armada-3720-espressobin-v7-emmc.dts b/arch/arm64/boot/dts/marvell/armada-3720-espressobin-v7-emmc.dts
index 3afe25484d..dabfedf813 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-espressobin-v7-emmc.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-espressobin-v7-emmc.dts
@@ -18,21 +18,28 @@
 
 	aliases {
 		/* ethernet1 is wan port */
-		ethernet1 = &switch0port3;
-		ethernet3 = &switch0port1;
+		ethernet1 = &switch0port1;
+		ethernet2 = &switch0port2;
+		ethernet3 = &switch0port3;
 	};
 };
 
 &ports {
 	switch0port1: port@1 {
 		reg = <1>;
-		label = "lan1";
+		label = "eth1";
 		phy-handle = <&switch0phy0>;
 	};
 
+	switch0port2: port@2 {
+		reg = <2>;
+		label = "eth2";
+		phy-handle = <&switch0phy1>;
+	};
+
 	switch0port3: port@3 {
 		reg = <3>;
-		label = "wan";
+		label = "eth3";
 		phy-handle = <&switch0phy2>;
 	};
 };
